### PR TITLE
ContestDetailViewModel unit test

### LINKED ISSUE #179: ContestDetailViewModel unit test
**Expected Behaviour**

Write ViewModel tests for ContestDetailViewModel

**Parent issue**
#60

### PR DESCRIPTION
**Pull request by jguerrerope**
Fixes #179 

**Changes:** 

- added library for arch unit test and test for CodingCalendarViewModel

### CODE REVIEW

File: app/src/test/java/com/iitp/njack/iitp_connect/core/calendar/detail/ContestDetailViewModelTest.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    public void shouldLoadContest_whenCurrentContestIsNull() {
+        // when
+        contestDetailViewModel.getContest(MOCK_CONTEST_ID);
+
+        // then
+        Mockito.verify(contestDetailRepository).getContest(Mockito.eq(MOCK_CONTEST_ID));
+    }
+
+    @Test
+    public void shouldLoadContest_whenCurrentContestIsNotNull() {
```
Comment (dr0pdb) [inline app/src/test/java/com/iitp/njack/iitp_connect/core/calendar/detail/ContestDetailViewModelTest.java]: Please use Camel Casing.

File: app/src/test/java/com/iitp/njack/iitp_connect/core/calendar/detail/ContestDetailViewModelTest.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    public InstantTaskExecutorRule ruleExecutorRule = new InstantTaskExecutorRule();
+
+    @InjectMocks
+    private ContestDetailViewModel contestDetailViewModel;
+
+    @Mock
+    private ContestDetailRepository contestDetailRepository;
+
+    @Test
+    public void shouldLoadContest_whenCurrentContestIsNull() {
```
Comment (dr0pdb) [inline app/src/test/java/com/iitp/njack/iitp_connect/core/calendar/detail/ContestDetailViewModelTest.java]: Camel casing please.

File: app/src/test/java/com/iitp/njack/iitp_connect/core/calendar/detail/ContestDetailViewModelTest.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    @Test
+    public void shouldLoadContest_whenCurrentContestIsNotNull() {
+        // given
+        MutableLiveData<Contest> contestLiveData = new MutableLiveData<>();
+        contestLiveData.setValue(MOCK_CONTEST);
+
+        Mockito.when(contestDetailRepository.getContest(MOCK_CONTEST_ID))
+            .thenReturn(contestLiveData);
+
+
```
Comment (dr0pdb) [inline app/src/test/java/com/iitp/njack/iitp_connect/core/calendar/detail/ContestDetailViewModelTest.java]: remove one blank line.

**Review by dr0pdb — CHANGES_REQUESTED** [review]

File: app/src/test/java/com/iitp/njack/iitp_connect/core/calendar/detail/ContestDetailViewModelTest.java (hunk 1 above)
Comment (jguerrerope) [reply app/src/test/java/com/iitp/njack/iitp_connect/core/calendar/detail/ContestDetailViewModelTest.java]: https://dzone.com/articles/7-popular-unit-test-naming

File: app/src/test/java/com/iitp/njack/iitp_connect/core/calendar/detail/ContestDetailViewModelTest.java (hunk 2 above)
Comment (jguerrerope) [reply app/src/test/java/com/iitp/njack/iitp_connect/core/calendar/detail/ContestDetailViewModelTest.java]: https://dzone.com/articles/7-popular-unit-test-naming

Comment (dr0pdb) [reply app/src/test/java/com/iitp/njack/iitp_connect/core/calendar/detail/ContestDetailViewModelTest.java]: Yes, I am asking you to do it like the point 3. `shouldLoadContestWhenCurrentContestIsNull`.

**Comment by jguerrerope** [comment]
updated.

**Review by dr0pdb — APPROVED** [review]

**Pull request merged**